File: ru/javastudy/main/homework5/FirstTask.java
package ru.javastudy.homework5;

import java.util.Arrays;

public class FirstTask {

    public static void main(String[] args) {
        Integer[] nums = { 1, 2, 3, 4, 5 };

        swapElements(nums, 2, 4);

        System.out.println(Arrays.toString(nums));
    }

    public static <T> void swapElements(T[] array, int in1, int in2) {
        if (
                array == null
                || in1 < 0
                || in1 > array.length
                || in2 < 0
                || in2 > array.length
        ) throw new IllegalArgumentException();

        T arr = array[in1];

        array[in1] = array[in2];
        array[in2] = arr;
    }

}

File: main/homework5/FirstTask.kt
package ru.kotlinstudy.main.homework5

fun main() {
    val strings = arrayOf(1, 2, 3, 4, 5)
    strings.swapElements(0, 2)
    println(strings.contentToString())
}

fun <T> Array<T>.swapElements(in1: Int, in2: Int) {
    if (in1 !in indices || in2 !in indices) throw IllegalArgumentException("Индексы выходят за границы массива")

    val temp = this[in1]
    this[in1] = this[in2]
    this[in2] = temp
}